### PR TITLE
pkg/asset/machines: Add load balancer refs to master machines on AWS

### PR DESCRIPTION
**Pull request by csrwng**
Pulls the latest code from cluster-api-aws-provider and sets up publicIP and LoadBalancers on cluster machines.

### CODE REVIEW

File: pkg/asset/machines/aws/machines.go [hunk 1]
```diff
@@ -98,7 +98,22 @@ func provider(clusterID, clusterName string, platform *aws.Platform, mpool *aws.
 				Values: []string{fmt.Sprintf("%s_%s_sg", clusterName, role)},
 			}},
 		}},
-	}, nil
+	}
+	if role == "master" {
+		public := true
+		providerConfig.PublicIP = &public
```
Comment (abhinavdahiya) [inline pkg/asset/machines/aws/machines.go]: we have `k8s.io/util/pointer` vendored.

Comment (csrwng) [reply pkg/asset/machines/aws/machines.go]: thx, fixed

File: pkg/asset/machines/aws/machines.go [hunk 2]
```diff
@@ -98,7 +98,22 @@ func provider(clusterID, clusterName string, platform *aws.Platform, mpool *aws.
 				Values: []string{fmt.Sprintf("%s_%s_sg", clusterName, role)},
 			}},
 		}},
-	}, nil
+	}
+	if role == "master" {
```
[the remark this answers is not in the document]
Comment (csrwng) [reply pkg/asset/machines/aws/machines.go]: @abhinavdahiya not sure what you mean... are you asking if it should be in its own function that's not common with workers?

[the remark this answers is not in the document]
Comment (abhinavdahiya) [reply pkg/asset/machines/aws/machines.go]: we do this here https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/asset/machines/master.go#L83-L87

File: Gopkg.lock [hunk 3]
```diff
@@ -224,7 +224,6 @@
   revision = "0e69bafe1d790b820cef859dd0126893bdbd53ad"
 
 [[projects]]
-  branch = "master"
```
Comment (wking) [inline Gopkg.lock]: Do you know what's going on here?  I don't understand this removal.  Also, can you spit out the vendor bump from the human-generated changes.  For example, see ac11b74 and a14da04 in #682.

File: pkg/asset/machines/aws/machines.go [hunk 4]
```diff
@@ -98,7 +99,21 @@ func provider(clusterID, clusterName string, platform *aws.Platform, mpool *aws.
...
+		providerConfig.PublicIP = pointer.BoolPtr(true)
```
Comment (wking) [inline pkg/asset/machines/aws/machines.go]: Do we want to hard-code this?  It currently matches [this default][1] from among [these choices][2].  But this code is far from that code.  Can we push this setting up into a shared type ([`aws.MachinePool`][3]?) to stay DRY?

[1]: https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/tfvars/tfvars.go#L95
[2]: https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/tfvars/aws/aws.go#L7-L12
[3]: https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/types/aws/machinepool.go#L3-L22

Comment (abhinavdahiya) [reply pkg/asset/machines/aws/machines.go]: i would not expose that in MachinePool and move
https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/tfvars/tfvars.go#L95 to be driven by Machines and MachineSets...

File: Gopkg.lock (hunk 3 above)
Comment (csrwng) [reply Gopkg.lock]: @wking ran `dep ensure -update sigs.k8s.io/cluster-api-provider-aws` again to split out the commit and now don't see this change anymore. Not sure why the difference.

File: pkg/asset/machines/aws/machines.go (hunk 4 above)
Comment (csrwng) [reply pkg/asset/machines/aws/machines.go]: @abhinavdahiya do you mean that I should make machines.Master a dependency of TerraformVariables and pass in the Machine object to TFVars ?

File: pkg/asset/machines/aws/machines.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (csrwng) [reply pkg/asset/machines/aws/machines.go]: got it, so you want the master-specific part to be placed in master.go

[the remark this answers is not in the document]
Comment (abhinavdahiya) [reply pkg/asset/machines/aws/machines.go]: Yeah makes most sense. But also we need take into account that hive should be able to apply those changes too.

File: pkg/asset/machines/aws/machines.go (hunk 4 above)
Comment (abhinavdahiya) [reply pkg/asset/machines/aws/machines.go]: > @abhinavdahiya do you mean that I should make machines.Master a dependency of TerraformVariables and pass in the Machine object to TFVars ?

Yeah, but not in this PR, i don't think we need to do that now.

**Comment by csrwng** [comment]
@abhinavdahiya @wking addressed your comments

**Comment by wking** [comment]
This looks good to me, but I'll wait to give @abhinavdahiya another look at it to.

What does this actually do?  We set up the load-balancer associations ourselves, so is this just keeping new cluster-api-aws-provider from stomping on that and pulling us out of those load balancers?  Or making newer cluster-api-aws-provider happier when they notice that we are attached to those load balancers?  Or setting the stage for masters that are created via the cluster API?  Or...?

**Pull request merged**